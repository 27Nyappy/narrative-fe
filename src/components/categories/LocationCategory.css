.LocationCategory {
  background-color: palegreen;
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  h2, div {
    align-self: center;
  }
}

.LocationCategoryContainer {
  width: 85%;
  height: 40%;
  display: flex;
  justify-content: space-between;
}