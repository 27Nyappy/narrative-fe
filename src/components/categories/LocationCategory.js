import React from 'react';
import PropTypes from 'prop-types';
import AddButton from './AddButton';
import ViewAllButton from './ViewAllButton';
import styles from './LocationCategory.css';

const LocationCategory = ({  handleAddLocation, handleViewLocations }) => {
  return (
    <section className={styles.LocationCategory}>
      <h2>Locations</h2>

      <div>
        <AddButton handleAddClick={handleAddLocation} />
        <ViewAllButton handleViewClick={handleViewLocations} />
      </div>
    </section>
  );
};

LocationCategory.propTypes = {
  handleAddLocation: PropTypes.func.isRequired,
  handleViewLocations: PropTypes.func.isRequired
};

export default LocationCategory;