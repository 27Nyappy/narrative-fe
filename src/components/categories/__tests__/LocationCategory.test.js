import React from 'react';
import { shallow } from 'enzyme';
import LocationCategory from '../LocationCategory';

describe('LocationCategory component', () => {
  it('renders a location category', () => {
    const wrapper = shallow(<LocationCategory handleAddLocation={() => true} handleViewLocations={() => true}/>);
    expect(wrapper).toMatchSnapshot();
  });
});