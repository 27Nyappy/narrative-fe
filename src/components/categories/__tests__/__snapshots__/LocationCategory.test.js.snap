// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCategory component renders a location category 1`] = `
<section>
  <h2>
    Locations
  </h2>
  <div>
    <AddButton
      handleAddClick={[Function]}
    />
    <ViewAllButton
      handleViewClick={[Function]}
    />
  </div>
</section>
`;
